### PR TITLE
Refactored the config for fine-tuning

### PR DESCRIPTION
**Pull request by cwognum**
## Changelogs

- The main script to run has been moved to [`graphium/cli/hydra.py`](https://github.com/WenkelF/graphium/blob/explore_finetuning_cas/graphium/cli/hydra.py). I also added this to the [`pyproject.toml`](https://github.com/WenkelF/graphium/blob/explore_finetuning_cas/pyproject.toml#L65-L67), so now you can run the main script by simply running `graphium-hydra`.
- The main script fixes all of https://github.com/datamol-io/graphium/issues/425. Notably, it adds a new [`wandb` key](https://github.com/WenkelF/graphium/blob/explore_finetuning_cas/expts/hydra-configs/finetuning/admet.yaml#L23-L27) to the `constants` by which you can enable and configure the Weights & Biases logger. 
- The main script can be used for both training and fine-tuning. To use it for finetuning, you need to add the [`finetuning.yaml` config](https://github.com/WenkelF/graphium/blob/explore_finetuning_cas/expts/hydra-configs/finetuning/admet.yaml).
```bash
graphium-hydra +finetuning=admet
```

- [`graphium/cli/finetune`](https://github.com/WenkelF/graphium/blob/explore_finetuning_cas/graphium/cli/finetune.py) adds an easy way to fine-tune on all the benchmarks for a single model in sequence. See `graphium finetune admet --help`. 
- Finetuning for now will only work for the ADMET tasks, because we use the a [filter function](https://github.com/WenkelF/graphium/blob/explore_finetuning_cas/graphium/finetuning/utils.py#L9-L37) to filter a big config with all tasks into a smaller config for a single task based on the [`task` key](https://github.com/WenkelF/graphium/blob/explore_finetuning_cas/expts/hydra-configs/finetuning/admet.yaml#L42). This is easy to change in the future (and I left a comment to indicate this!)
- Since the main script now does everything, I [deprecated](https://github.com/WenkelF/graphium/blob/explore_finetuning_cas/expts/main_run_multitask.py#L35-L37) the old one.

### CODE REVIEW

File: expts/hydra-configs/tasks/admet.yaml [hunk 1]
```diff
@@ -1,128 +1,6 @@
...
-      hidden_dims: *gnn_dim
-      depth: 1
-      activation: relu
-      last_activation: none
-      dropout: *dropout
-      normalization: *normalization
-      last_normalization: "none"
-      residual_type: none
-
   task_heads:
```
Comment (cwognum) [inline expts/hydra-configs/tasks/admet.yaml]: @WenkelF One question that came up: Is it a problem that the fine-tuning config now overrides the task-heads or is that meant to happen? If it is a problem, let me know and I will fix it.

**Comment by cwognum** [comment]
I'll have to fix the failing test cases! In the meantime, I will convert this to a draft. Please don't merge yet!

File: expts/hydra-configs/tasks/admet.yaml (hunk 1 above)
Comment (cwognum) [reply expts/hydra-configs/tasks/admet.yaml]: I ended up changing this so we don't overwrite the task heads of the pre-trained model. I figured this was a bug.

**Comment by WenkelF** [comment]
Thanks @cwognum,  it looks great!

**Pull request merged**